pkg/controller/core.oam.dev/v1alpha2/application/application_controller.go

```diff
@@ -0,0 +1,182 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package application
+
+import (
+	"context"
+
+	"github.com/crossplane/crossplane-runtime/pkg/logging"
+
+	"github.com/oam-dev/kubevela/apis/core.oam.dev/v1alpha2"
+	core "github.com/oam-dev/kubevela/pkg/controller/core.oam.dev"
+
+	"github.com/pkg/errors"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+
+	"github.com/oam-dev/kubevela/pkg/controller/core.oam.dev/v1alpha2/application/builder"
+	fclient "github.com/oam-dev/kubevela/pkg/controller/core.oam.dev/v1alpha2/application/defclient"
+	"github.com/oam-dev/kubevela/pkg/controller/core.oam.dev/v1alpha2/application/parser"
+	"github.com/oam-dev/kubevela/pkg/controller/core.oam.dev/v1alpha2/application/template"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// ApplicationReconciler reconciles a Application object
+type applicationReconciler struct {
+	client.Client
+	Log    logr.Logger
+	Scheme *runtime.Scheme
+}
+
+// +kubebuilder:rbac:groups=core.oam.dev,resources=applications,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=core.oam.dev,resources=applications/status,verbs=get;update;patch
+
+// Reconcile process app event
+func (r *applicationReconciler) Reconcile(req ctrl.Request) (result ctrl.Result, gerr error) {
+
+	ctx := context.Background()
+	_log := r.Log.WithValues("application", req.NamespacedName)
```

Comment:
we usually not use underscore to name variable, I suggest use formal one like:
```suggestion
	appLog := r.Log.WithValues("application", req.NamespacedName)
```